src/databricks/labs/mcp/servers/unity_catalog/tools/vector_search.py

```diff
@@ -1,62 +1,78 @@
-import io
 import json
-from contextlib import redirect_stdout
+from typing import List
 
-from databricks_openai import VectorSearchRetrieverTool
+from pydantic import BaseModel
 from databricks.sdk import WorkspaceClient
+from databricks.vector_search.client import VectorSearchClient
+from databricks.labs.mcp.servers.unity_catalog.tools.base_tool import BaseTool
+from mcp.types import TextContent, Tool as ToolSpec
 
-from mcp.types import Tool as ToolSpec, TextContent
 
-from databricks.labs.mcp.servers.unity_catalog.tools.base_tool import BaseTool
+class QueryInput(BaseModel):
+    query: str
 
 
 class VectorSearchTool(BaseTool):
-    def __init__(self, tool_obj: VectorSearchRetrieverTool):
-        self.tool_obj = tool_obj
-        tool_info = tool_obj.tool["function"]
-        llm_friendly_tool_name = tool_info["name"]
+    def __init__(self, endpoint_name: str, index_name: str, tool_name: str, columns: List[str]):
+        self.endpoint_name = endpoint_name
+        self.index_name = index_name
+        self.tool_name = tool_name
+        self.columns = columns
+
         tool_spec = ToolSpec(
-            name=llm_friendly_tool_name,
-            description=tool_info["description"],
-            inputSchema=tool_info["parameters"],
+            name=tool_name,
+            description=f"Searches the vector index `{index_name}`.",
+            inputSchema=QueryInput.model_json_schema(),
         )
-        super().__init__(tool_spec=tool_spec)
+        super().__init__(tool_spec)
 
     def execute(self, **kwargs):
-        """
-        Executes the vector search tool with the provided arguments.
-        """
-        # Create a buffer to capture stdout from vector search client
-        # print statements
-        f = io.StringIO()
-        with redirect_stdout(f):
-            res = self.tool_obj.execute(**kwargs)
-            return [
-                TextContent(
-                    type="text",
-                    text=json.dumps(vs_res),
-                )
-                for vs_res in res
-            ]
+        model = QueryInput.model_validate(kwargs)
+        vsc = VectorSearchClient(disable_notice=True)
+
+        index = vsc.get_index(index_name=self.index_name)
+
+        results = index.similarity_search(
+            query_text=model.query,
+            columns=self.columns,
+            num_results=5,
```

Comment:
can this be a parameter picked up from global settings and passed into `__init__` ?